client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +49,61 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final boolean subscriberForRetention;
+    private final boolean autoPublishAtLastCheckpoint;
+
+    public enum ReaderGroupRetentionConfig {
+        /**
+         * There is no retention done by this {@link ReaderGroup}.
+         */
+        NO_CONSUMPTION_BASED_TRUNCATION,
+
+        /**
+         * This {@link ReaderGroup} can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream according to its read positions. These read positions have to be provided
+         * manually in the form of a {@link StreamCut}, see this method:
+         *
+         * {@link ReaderGroup#updateRetentionStreamCut(Stream, StreamCut)}
+         */
+        TRUNCATE_AT_USER_STREAMCUT,
+
+        /**
+         * This {@link ReaderGroup} can retain stream data using Consumption Based Retention. This will configure the
+         * {@link ReaderGroup} to truncate the stream using the read positions from the {@link StreamCut}s generated during checkpointing.
+         */
+        TRUNCATE_AT_LAST_CHECKPOINT
+    }
+
    public static class ReaderGroupConfigBuilder implements ObjectBuilder<ReaderGroupConfig> {
        private long groupRefreshTimeMillis = 3000; //default value
        private long automaticCheckpointIntervalMillis = 30000; //default value
        // maximum outstanding checkpoint request that is allowed at any given time.
        private int maxOutstandingCheckpointRequest = 3; //default value
+       private boolean subscriberForRetention = false; //default value
```

Comment:
Please remove these 2 flags from here and add them to the ReaderGroupRetentionConfig  ENUM. ENUMs in java can have values.

Reply:
Something like:
```
enum ReaderGroupRetentionConfig {
        NO_CONSUMPTION_BASED_TRUNCATION(false, false),
        TRUNCATE_AT_USER_STREAMCUT(true, false),
        TRUNCATE_AT_LAST_CHECKPOINT(true, true);
        
        private boolean isReaderGroupASubscriber;
        private boolean autoTruncateAtLastCheckpoint;
        
        ReaderGroupRetentionConfig(boolean isSubscriber, boolean autoTruncateAtLastCheckpoint)
        {
            this.isReaderGroupASubscriber = isSubscriber;
            this.autoTruncateAtLastCheckpoint = autoTruncateAtLastCheckpoint;
        }
    }
```

Reply:
Implemented as described.